### PR TITLE
Provide IFR data from package

### PR DESCRIPTION
**Pull request by pratikunterwegs**
This PR makes the infection IFR values available for the DAEDALUS-relevant age group in `infection_data` and in `<daedalus_infection>` objects under the name `"ifr"`. I haven't added any specific tests as most of this should be covered by class validation.

How to use:

``` r
x = daedalus::daedalus_infection("sars_cov_1")
y = daedalus::daedalus_country("China")

# simple mean
mean(x$ifr)
#> [1] 0.07544104

# weighted mean
weighted.mean(
  x$ifr, y$demography
)
#> [1] 0.04680522
```

<sup>Created on 2024-10-25 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

### CODE REVIEW

File: data-raw/infection_data.R [hunk 1]
```diff
@@ -168,8 +169,8 @@ infection_data <- Map(
 
 usethis::use_data(infection_data, overwrite = TRUE)
 
-# save infection parameter names
-infection_parameter_names <- names(infection_data)
+# save infection parameter names, access first element
+infection_parameter_names <- names(infection_data[[1L]])
```
Comment (EmmaLRussell) [inline data-raw/infection_data.R:173]: Why was this change required?

**Review by EmmaLRussell — APPROVED** [review]
Looks good, as far as I can tell!

File: data-raw/infection_data.R (hunk 1 above)
Comment (pratikunterwegs) [reply data-raw/infection_data.R:173]: I must have had this wrong earlier, but the names of `infection_data` are the epidemic names, and not the parameter names. The infection parameters are nested within each infection/epidemic list, so this get those.
